porcupine/plugins/git_status.py

```diff
@@ -78,34 +78,41 @@ def __init__(self, tree: DirectoryTree, project_id: str):
         self.tree = tree
         self.project_id = project_id
         self.project_path = get_path(project_id)
-        self.git_status_future: Future[dict[Path, str]] | None = None
         self.queue: set[str] = set()
+        self._git_status_future: Future[dict[Path, str]] | None = None
 
     def start_running_git_status(self) -> None:
-        self.git_status_future = git_pool.submit(partial(run_git_status, self.project_path))
+        self._git_status_future = git_pool.submit(partial(run_git_status, self.project_path))
 
         # Handle queue contents when it has completed
         def check() -> None:
-            if self.git_status_future is not None:
-                if self.git_status_future.done():
+            if self._git_status_future is not None:
+                if self._git_status_future.done():
                     self._handle_queue()
                 else:
                     self.tree.after(25, check)
 
         check()
 
     def stop(self) -> None:
-        self.git_status_future = None
+        self._git_status_future = None
 
     def _choose_tag(self, item_path: Path) -> str | None:
         # process should be done, result available immediately
-        assert self.git_status_future is not None
-        path_to_status = self.git_status_future.result(timeout=0)
-
-        for path, status in path_to_status.items():
-            # use status of a folder also for its contents
-            if path == item_path or path in item_path.parents:
-                return status
+        assert self._git_status_future is not None
+        path_to_status = self._git_status_future.result(timeout=0)
+
+        # path_to_status can be huge if lots of ignored files, avoid looping over it
+        try:
+            return path_to_status[item_path]
+        except KeyError:
+            # status of a folder applies to everything it contains
+            for folder in item_path.parents:
+                try:
+                    return path_to_status[folder]
+                except KeyError:
+                    if folder == self.project_path:
+                        break
 
         # Handle directories containing files with different statuses
         substatuses = {
```

Comment:
The code below loops over all the statuses and still isn't really ideal. I'm starting to think that `dict[Path, status string]` is maybe not the ideal data structure for this,.